### PR TITLE
sca: codechecker: Twister default values

### PR DESCRIPTION
**Pull request by pdgendt**
This PR enables running Static Code Analysis tools as part of twister. The main focus is to have a [CodeChecker](https://codechecker.readthedocs.io/en/latest/) server running to track reported issues.

An example:
```shell
# Setup the environment
$ export CODECHECKER_CONFIG_FILE="$PWD/.codechecker.yml"
$ export CODECHECKER_STORE_OPTS="--url;localhost:8001/Zephyr"
$ export CODECHECKER_CLEANUP=y
$ export CODECHECKER_PARSE_SKIP=y
# Run some twisters
$ west twister -T samples/net/sockets -p native_sim --build-only -x=ZEPHYR_SCA_VARIANT=codechecker
$ west twister -T samples/hello_world -p mimxrt1064_evk -p qemu_x86 --build-only -x=ZEPHYR_SCA_VARIANT=codechecker
```

image
image
image

### CODE REVIEW

**Review by hakehuang — DISMISSED** [review]

File: doc/develop/sca/codechecker.rst [hunk 1]
```diff
@@ -33,56 +33,62 @@ called with a ``-DZEPHYR_SCA_VARIANT=codechecker`` parameter, e.g.
...
+
+These parameters can be passed on the command line, or be set as environment variables.
+
+
+Running twister with CodeChecker
+********************************
+
+When running ``twister``, it is possible to run CodeChecker static code analysis as well. Pass
+``--sca-variant codechecker`` when calling ``twister`` to enable it. Once enabled, some default
+options are set as following:
```
[the remark this answers is not in the document]
Comment (pdgendt) [reply doc/develop/sca/codechecker.rst:78]: Tried this, but referring from the docs:

```
Exit status
------------------------------------------------
0 - No report
1 - CodeChecker error
2 - At least one report emitted by an analyzer
```

There will almost always be some report, so not sure how useful this is in twister.
The same can be said for code coverage, should you "fail" if a coverage threshold isn't reached?

[the remark this answers is not in the document]
Comment (golowanow) [reply doc/develop/sca/codechecker.rst:78]: My point is to have the same behavior for all SCA in Twister with default options, but currently this PR results differently:
* with `--sca-variant=gcc` there are build warnings from SCA, eventually Twister test instance build error.
* with `--sca-variant=codechecker` issues become errors only with additional `CODECHECKER_PARSE_EXIT_STATUS=y`

This should be, at least, documented in the `--sca-variant` description.

File: cmake/sca/codechecker/sca.cmake [hunk 2]
```diff
@@ -5,6 +5,40 @@
 find_program(CODECHECKER_EXE NAMES CodeChecker codechecker REQUIRED)
 message(STATUS "Found SCA: CodeChecker (${CODECHECKER_EXE})")
 
+# Get CodeChecker specific variables
+zephyr_get(CODECHECKER_ANALYZE_JOBS)
```
Comment (tejlmand) [inline cmake/sca/codechecker/sca.cmake:9]: I know each SCA tool will have its own prefered way of being configured, but at the same time Zephyr users are familiar with Kconfig.

During review of Eclair integration it became clear that a better Kconfig integration to SCA tools is needed, see: #68324 and https://github.com/zephyrproject-rtos/zephyr/issues/77018, especially in a way that the SCA Kconfig is only loaded when the tool is available.

The list of settings below looks like something that would be better placed in Kconfig, and thereby also be auto documented using the Kconfig help text.

Perhaps with an exception for the `CODECHECKER_CONFIG_FILE` as that could look like something one might want to pass in directly on CMake invocation.

Thoughts ?

File: scripts/pylib/twister/twisterlib/sca.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# vim: set syntax=python ts=4 :
```
Comment (tejlmand) [inline scripts/pylib/twister/twisterlib/sca.py:1]: I don't think we generally add editor settings in files, although I give a :+1: for the choice of editor.

```suggestion
```

**Review by tejlmand — COMMENTED** [review]
Thanks for this.

Some comments, but so far no hard blockers.

File: scripts/pylib/twister/twisterlib/sca.py (hunk 3 above)
Comment (kartben) [reply scripts/pylib/twister/twisterlib/sca.py:1]: +1 - I think the accepted practice is just to run files through ``black file.py -l 100``

File: tests/codechecker_config.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2024, Basalte bv
```
Comment (nashif) [inline tests/codechecker_config.yaml:3]: i would put this file somewhere else, the config might be applied when building tests, but we are checking the zephyr code, not the tests only, maybe a top level dot file or directory with multiple configs?

**Review by nashif — APPROVED** [review]

**Review by hakehuang — APPROVED** [review]

**Pull request merged**